pkg/consolelink/main.go

```diff
@@ -0,0 +1,45 @@
+package consolelink
+
+import (
+	"fmt"
+
+	consolev1 "github.com/openshift/api/console/v1"
+	routev1 "github.com/openshift/api/route/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	v1 "github.com/jaegertracing/jaeger-operator/pkg/apis/jaegertracing/v1"
+)
+
+// RouteAnnotation used to annotate the link with the route name
+var RouteAnnotation = "consolelink.jaegertracing.io/route"
+
+// Get returns an ConsoleLink specification for the current instance
+func Get(jaeger *v1.Jaeger, route *routev1.Route) *consolev1.ConsoleLink {
+	return &consolev1.ConsoleLink{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: jaeger.Namespace + ".jaeger-consolelink-" + jaeger.Name,
+			Namespace:jaeger.Namespace,
+			Labels: map[string]string{
+				"app.kubernetes.io/instance":   jaeger.Name,
+				"app.kubernetes.io/managed-by": "jaeger-operator",
+			},
+			Annotations: map[string]string{
+				RouteAnnotation: route.Name,
+			},
+		},
+		Spec: consolev1.ConsoleLinkSpec{
```

Comment:
This is using a different structure to the examples of [kiali](https://github.com/kiali/kiali-operator/blob/v1.18.2/roles/default/kiali-deploy/templates/openshift/console-links.yaml#L13-L32) and the [blog](https://www.openshift.com/blog/openshift-4-3-the-project-launcher).

We need the console link to only display in the namespace where the Jaeger instance is deployed.

Reply:
Done